src/main/java/httpserver/httpmessages/HTTPRequestHandler.java

```diff
@@ -1,17 +1,54 @@
 package httpserver.httpmessages;
 
 import httpserver.RequestHandler;
-import httpserver.ResponseBuilder;
+import httpserver.ResourceHandler;
+import httpserver.resourcemanagement.Resource;
+import httpserver.routing.Router;
+
+import static httpserver.httpmessages.StatusCode.NOTALLOWED;
+import static httpserver.httpmessages.StatusCode.NOTFOUND;
+import static httpserver.httpmessages.StatusCode.OK;
+import static httpserver.routing.Method.GET;
 
 public class HTTPRequestHandler implements RequestHandler {
 
-    private final ResponseBuilder responseBuilder;
+    private final ResourceHandler resourceHandler;
+    private final Router router;
 
-    public HTTPRequestHandler(ResponseBuilder responseBuilder) {
-        this.responseBuilder = responseBuilder;
+    public HTTPRequestHandler(ResourceHandler resourceHandler, Router router) {
+        this.resourceHandler = resourceHandler;
+        this.router = router;
     }
 
     public HTTPResponse handle(HTTPRequest httpRequest) {
-        return responseBuilder.buildResponse(httpRequest.getMethod(), httpRequest.getRequestURI());
+        Resource resource = resourceHandler.getResource(httpRequest.getRequestURI());
+        if (resource.exists()) {
+            if (router.allowsMethod(httpRequest.getMethod())) {
+                HTTPResponse httpResponse = getOKResponse();
+                if (httpRequest.getMethod() == GET) {
+                    setBody(resource, httpResponse);
+                }
+                return httpResponse;
+            }
+            return new HTTPResponse(NOTALLOWED.code, NOTALLOWED.reason);
+        }
+        if (router.hasRegistered(httpRequest.getRequestURI())) {
+            return router.getResponse(httpRequest);
+        }
+        return getNotFoundResponse();
+    }
```

Comment:
I'm still figuring out how to refactor this, I agree it is complicated, and this whole class might not be that useful, but haven't quite figured out how I want to change it yet...